src/org/linphone/DialerFragment.java

```diff
@@ -298,8 +298,21 @@ public void onClick(View v) {
 
 		Log.d("mCamera" + mCamera);
 
-		mPreview = new CameraPreview(myContext, mCamera);
-		cameraPreview.addView(mPreview);
+		if(cameraPreview.getChildCount() == 1)
+		{
+			mPreview = (CameraPreview) cameraPreview.getChildAt(0);
```

Comment:
Should add sanity check here to evaluate whether cameraPreview.getChildAt(0) is equal to typeof(CameraPreview), if not this could crash if later on there were additional views.